src/Raven.Server/Utils/BackupHistoryWriter.cs

```diff
@@ -0,0 +1,241 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Raven.Client.Documents.Operations.Backups;
+using Raven.Server.Documents;
+using Raven.Server.Documents.PeriodicBackup.BackupHistory;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.ServerWide;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Utils
+{
+    public class BackupHistoryWriter
+    {
+        public const string IdPropertyName = "Id";
+        public const string TaskNamePropertyName = "TaskName";
+
+        private readonly string _currentNodeTag;
+        private readonly string _requestedNodeTag;
+        private readonly long? _requestedTaskId;
+        private readonly Dictionary<(string NodeTag, long TaskId), List<BlittableJsonReaderObject>> _clusterStorageEntries = new();
+        private readonly Dictionary<long, List<BlittableJsonReaderObject>> _temporaryStorageEntries = new();
+        private readonly HashSet<DateTime> _clusterEntriesCreationDateTimes = new();
+        private bool _isFirst = true;
+
+        public BackupHistoryWriter(string requestedNodeTag, long? requestedTaskId, string currentNodeTag)
+        {
+            _requestedNodeTag = requestedNodeTag;
+            _requestedTaskId = requestedTaskId;
+            _currentNodeTag = currentNodeTag;
+        }
+
+        public void GetBackupHistoryEntriesFromCluster(DocumentDatabase database, ServerStore serverStore, TransactionOperationContext context)
```

Comment:
ClusterStateMachine should have a property Backups where all the methods for retrieving, etc are there

Writing should be as simple as creating in the endpoint writer and processing items retrieved from CSM using .ToJson() method